src/hotspot/cpu/aarch64/assembler_aarch64.cpp

```diff
@@ -54,10 +54,32 @@ Assembler::SIMD_Arrangement Assembler::_esize2arrangement_table[9][2] = {
   /*   8  */      {T1D,                 T2D}
   };
 
+Assembler::SIMD_RegVariant Assembler::_esize2regvariant[9] = {
+  INVALID,
+  B,
+  H,
+  INVALID,
+  S,
+  INVALID,
+  INVALID,
+  INVALID,
+  D,
+};
+
+Assembler::SIMD_Arrangement Assembler::esize2arrangement(unsigned esize, bool isQ) {
+  assert(esize < ARRAY_SIZE(_esize2arrangement_table) &&
```

Comment:
Please consider making this a guarantee. The cost is minimal, and we've been saved by guarantees elsewhere in this file.

Reply:
Done.